test/package.json

```diff
@@ -1,14 +1,19 @@
 {
   "#": "We must put jest in a sub-directory otherwise VS Code somehow picks up the types and generates conflicts with mocha.",
+  "scripts": {
+    "postinstall": "./scripts/patch.sh"
+  },
   "devDependencies": {
     "@types/jest": "^26.0.20",
     "@types/jsdom": "^16.2.6",
     "@types/node-fetch": "^2.5.8",
+    "@types/service-worker-mock": "^2.0.1",
```

Comment:
Should we try mocking it ourselves to avoid the patching?

Something like this would work:

```typescript
interface Listener {
  event: string
  cb: (event?: { claim: () => void }) => void
}

const listeners: Listener[] = []

Object.assign(global, {
  self: global,
  addEventListener: (event: string, cb: () => void) => {
    listeners.push({ event, cb })
  },
  clients: {
    claim: () => undefined // Or a Jest spy
  },
})

function emit(event: string) {
  listeners.filter((l) => l.event === event).forEach((l) => {
    switch (event) {
      case "activate":
        l.cb({
          waitUntil: () => undefined // Or a Jest spy
        })
        break
      default:
        l.cb()
    }
  })
}
```

Reply:
Sure, it's worth trying! Oh perfect. I'll try this out.

Reply:
Wow...dude I didn't realize it was that easy to mock 😂 that worked like a charm ✨